lib/typecheck/gen_constraints.ml

```diff
@@ -22,11 +23,11 @@ module IEnv = Interface_env
 
 (* Synthesis mode *)
 let rec synthesise_val :
```

Comment:
We shouldn't need a positional argument to the typechecking functions, since they should be associated with the relevant AST node

Reply:
I’m not entirely sure I fully understand your point, but just to clarify, there are only two instances where positional information is explicitly passed in: one is during the `synthesise_val`, and the other is `check_val`.